### DISCUSSION TITLE
highway-v0 environment Multiple controlled vehicle

### DISCUSSION THREAD

**Discussion by yubai111** [eleurent/highway-env#157] [Q&A]
Thank you for creating such a perfect environment for autonomous driving. In the example you give, a vehicle can run reinforcement learning perfectly and avoid obstacles perfectly. But wheni changed ‘controlled vehicles’ parameter as 2, one of the vehicles was out of control. I don't know why. Could you teach me?

**Comment by eleurent** [comment]
I'm not sure why, what kind of RL algorithm are you using? Does your implementation support the multi-agent setting?

**Reply by yubai111** [reply]
i used the demo that you gave in goole colab , because i am a fresh . i dont know how to support the multi-agent setting

**Reply by 1028686314** [reply]
Is there a link to the code for this demo?

**Reply by eleurent** [reply]
You can run [this script](https://github.com/eleurent/rl-agents/blob/master/scripts/experiments.py) with the following arguments:
```
cd /path/to/rl-agents/scripts/
python experiments.py evaluate configs/IntersectionEnv/env_multi_agent.json  configs/IntersectionEnv/agents/DQNAgent/ego_attention_2h.json --train --episodes=3000
```

**Reply by eleurent** [reply]
Alternatively if you are running [this colab](https://colab.research.google.com/github/eleurent/highway-env/blob/master/scripts/intersection_social_dqn.ipynb#scrollTo=QowKW3ix45ZW), you can just replace the line

```env_config = 'configs/IntersectionEnv/env.json'```

by

```env_config = 'configs/IntersectionEnv/env_multi_agent.json'```
